xcell/mappers/mapper_KV450.py

```diff
@@ -141,12 +147,33 @@ def _remove_multiplicative_bias(self, cat_data, zbin):
         cat_data['bias_corrected_e2'][sel_gals] /= 1 + self.m[zbin]
 
     def _get_gals_or_stars(self, kind='galaxies'):
-        sel = self.cat['SG_FLAG'] == self.sel[kind]
-        return self.cat[sel]
+        self.cat_data = self.get_catalog()
+        sel = self.cat_data['SG_FLAG'] == self.sel[kind]
+        return self.cat_data[sel]
 
     def get_signal_map(self, mode=None):
         kind, e1f, e2f, mod = self._set_mode(mode)
-        if self.maps[mod] is None:
+        if self.maps[mod] is not None:
+            self.signal_map = self.maps[mod]
+            return self.signal_map
+
+        # This will only be computed if self.maps['mod'] is None
+        file_name1 = \
+            f'KV450_signal_map_{mod}_e1_zbin{self.zbin}_ns{self.nside}'
+        file_name2 = \
+            f'KV450_signal_map_{mod}_e2_zbin{self.zbin}_ns{self.nside}'
+        # Add extension separately to respect text width < 78
+        file_name1 += '.fits.gz'
+        file_name2 += '.fits.gz'
+        read_lite1, fname_lite1 = self._check_lite_exists(file_name1)
+        read_lite2, fname_lite2 = self._check_lite_exists(file_name2)
+        if read_lite1 and read_lite2:
+            print('Loading bin{} signal map'.format(self.zbin))
+            e1 = hp.read_map(fname_lite1)
+            e2 = hp.read_map(fname_lite2)
+            self.maps[mod] = [-e1, e2]
+        else:
+            print('Computing bin{} signal map'.format(self.zbin))
```

Comment:
You're not writing the maps after computing them

Reply:
Thank u for spotting these. Now it should all be fine.